cpp/FileTransferAgent.cpp

```diff
@@ -251,16 +257,23 @@ void Snowflake::Client::FileTransferAgent::uploadFilesInParallel(std::string *co
     tp.AddJob([metadata, resultIndex, command, this]()->void {
         do
         {
+          CXX_LOG_DEBUG("Putget Parallel upload %s", metadata->srcFileName.c_str());
           RemoteStorageRequestOutcome outcome = uploadSingleFile(m_storageClient, metadata,
                                                      resultIndex);
           if (outcome == RemoteStorageRequestOutcome::TOKEN_EXPIRED)
           {
+            CXX_LOG_DEBUG("Token expired, Renewing token.")
             _mutex_lock(&m_parallelTokRenewMutex);
             this->renewToken(command);
             _mutex_unlock(&m_parallelTokRenewMutex);
           }
           else
           {
+            if( outcome == RemoteStorageRequestOutcome::FAILED)
+            {
+              throw SnowflakeTransferException(TransferError::INTERNAL_ERROR,
```

Comment:
Again, add more information to the user-visible error message

Reply:
Yes, will add the reason for failure and the file name.